pymc/printing.py

```diff
@@ -26,91 +28,61 @@
 from pymc.model import Model
 
-__all__ = [
-    "str_for_dist",
-    "str_for_model",
-    "str_for_potential_or_deterministic",
-]
+__all__ = ["str_for_model", "str_for_model_var"]
 
 
-def str_for_dist(
-    dist: TensorVariable, formatting: str = "plain", include_params: bool = True
+def str_for_model_var(
+    var: TensorVariable, formatting: str = "plain", dist_name: Optional[str] = None, **kwargs
 ) -> str:
-    """Make a human-readable string representation of a Distribution in a model, either
-    LaTeX or plain, optionally with distribution parameter values included."""
-
-    if include_params:
-        # first 3 args are always (rng, size, dtype), rest is relevant for distribution
-        if isinstance(dist.owner.op, RandomVariable):
-            dist_args = [
-                _str_for_input_var(x, formatting=formatting) for x in dist.owner.inputs[3:]
-            ]
-        else:
-            dist_args = [
-                _str_for_input_var(x, formatting=formatting)
-                for x in dist.owner.inputs
-                if not isinstance(x, (RandomStateSharedVariable, RandomGeneratorSharedVariable))
-            ]
-
-    print_name = dist.name
-
-    if "latex" in formatting:
-        if print_name is not None:
-            print_name = r"\text{" + _latex_escape(dist.name.strip("$")) + "}"
-
-        op_name = (
-            dist.owner.op._print_name[1]
-            if hasattr(dist.owner.op, "_print_name")
-            else r"\\operatorname{Unknown}"
+    """Make a human-readable string representation of a Model variable.
+
+    Intended for Distribution, Deterministic, and Potential.
+    """
+    if not (
+        _has_owner(var) and isinstance(var.owner.op, (RandomVariable, SymbolicRandomVariable))
+    ) and not _is_potential_or_deterministic(var):
+        raise ValueError(
+            f"Variable for pretty-printing must be a model variable or the output of .dist(). Received unsupported variable {var}"
         )
-        if include_params:
-            if print_name:
-                return r"${} \sim {}({})$".format(
-                    print_name, op_name, ",~".join([d.strip("$") for d in dist_args])
-                )
-            else:
-                return r"${}({})$".format(op_name, ",~".join([d.strip("$") for d in dist_args]))
-
-        else:
-            if print_name:
-                return rf"${print_name} \sim {op_name}$"
-            else:
-                return rf"${op_name}$"
-
-    else:  # plain
-        dist_name = (
-            dist.owner.op._print_name[0] if hasattr(dist.owner.op, "_print_name") else "Unknown"
+    var_name, dist_name, args_str = _get_varname_distname_args(
+        var, formatting=formatting, dist_name=dist_name
+    )
+    if "include_params" in kwargs:
+        warnings.warn(
+            "The `include_params` argument has been deprecated. Future versions will always include parameter values.",
+            FutureWarning,
+            stacklevel=2,
```

Comment:
Your point of depreciating `include_params` seems valid to me. If IIRC when browsing this file a few weeks or months ago, `include_params` was used to delineate whether they were printed as arguments to other distributions or not. You seem to have addressed this in `_get_varname_distname_args`.